### ISSUE TITLE
Improve search results and add Top 20 Most Popular tokens

### ISSUE THREAD

**Issue by tomekalphawallet** [AlphaWallet/alpha-wallet-android#1570]
**Problem:**
During user tests, I discover that users have problems understanding how the wallet works. They think if something is not listed in the wallet tab - it is not available. And even if they try to look for a specific coin in the Add/Hide Tokens - if they can't search or find tokens they want to transfer to the wallet - they assume these tokens are not available for this wallet.

One user thought that a coin is not available because it was not listed in the wallet tab. So there was a wrong assumption that if something was not visible, it must be unavailable.

**Solution:**

1. Below "Hidden Tokens" add another section called "Popular Tokens" where we list the top 20 most popular erc20 tokens.
https://zpl.io/VKMMAjX
2. The Search box in Add/Hide Tokens should display (as a result of searching) even those tokens that are out of the wallet, but still, ERC20.
https://zpl.io/VKMMAjX https://zpl.io/aNXXGjn
3. Add Search bar (working the same way as 2) to the Add Custom Token screen.
https://zpl.io/beRRBZP

Not related but can be updated along with this change:
4. Remove Plus icon symbol from the wallet (leave just a text "Add/Hide Tokens"
https://zpl.io/aRdOwWK
5. Make sure that the change from https://github.com/AlphaWallet/alpha-wallet-android/issues/1487 is also applied.
6. You can remove "Hide Zero Balance Tokens" Feature

![Add_Hide_Tokens_Android](https://user-images.githubusercontent.com/51817359/90648283-9cce4b00-e239-11ea-8da6-c05cde99de2c.jpg)

**Comment by AW-STJ** [comment]
Great designs.

@ChintanRathod : Can you provide me a very high level rought estimate of time required to do this piece of work? Thinking of getting a Beta version out for testing and then we can refine it before getting it into production

**Comment by JamesSmartCell** [comment]
The approach should be like this:

- add the required contracts to known_contracts array in the resources. There is already a mechanism to discover these token details which fires when the new account is created.
- On startup of the add/hide tokens screen, check the contents of the known_contracts, resolve them from the TokenService and display the ones that haven't been specifically selected as 'popular tokens'.
- If there are tokens on the list that haven't been discovered yet (presumably the user went straight to add/hide tokens when creating their wallet), then add a token listener similar to that in WalletFragment which will populate the tokens as they become available (no need to hold up the view being drawn to start with). For bonus points you can add placeholders for those tokens, showing only the address so the user will see some data is still being fetched.

@tomekalphawallet or @AW-STJ can you supply a list of the required contracts for popular tokens? A good starting place would be looking for ERC20 in CoinMarketCap.

@ChintanRathod you can test already with the contracts already in the list. Once working this list can be added to on an ad-hoc basis by anyone in the team when they want to show a new token.

**Comment by ChintanRathod** [comment]
I understand your flow and points. As you mentioned, I saw "kwown_contract" list which mentioned with MainNet and xDAI for now.

**Comment by ChintanRathod** [comment]
> Great designs.
> 
> @ChintanRathod : Can you provide me a very high level rought estimate of time required to do this piece of work? Thinking of getting a Beta version out for testing and then we can refine it before getting it into production

Not sure about full implementation, but if I check points mentioned by James, covering 1 and 2 will be easy. 3rd will be time consuming. From my point, it would take 1 MD.

**Comment by JamesSmartCell** [comment]
@tomekalphawallet "You can remove the hide zero balance tokens" ? This is an important feature for removing wallet clutter. Would you say the default will be that all zero balance tokens are hidden?

**Comment by AW-STJ** [comment]
I would think yes, atleast for any tokens that:
1.  has not had any transactions 
2.  was not added manually by the users

Will differ to Tomek for the final decision.

**Comment by JamesSmartCell** [comment]
Thanks Sunil, I made a small modification to the algorithm to provide the best experience:

1. On startup, the Popular tokens are added to a watch list. They are all disabled unless they have a balance, in which case they are shown.
2. If the user changes the visibility of any token, this takes priority over anything else.

Bascially: if the wallet sees a token balance for a popular token, it will show the token balance. I checked with some whale wallets and it appears to work correctly.

If we're happy with this discovery algorithm then I'm OK with the 'show zero balance' option being removed.
